### PR TITLE
[4.0] Load showon js only once

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
Take loading the script for showon fields out of the foreach loop 

### Testing Instructions
Go to a view with showon fields, i.e. com_config. 
Showon fields must work before and after the patch.

### CODE REVIEW

**Comment by brianteeman** [comment]
aren't you now loading it even if there are no showon fields?

**Comment by chmst** [comment]
Yes, but in every view is at least one. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27963">issues.joomla.org/tracker/joomla-cms/27963</a>.</sub>

**Comment by richard67** [comment]
> Yes, but in every view is at least one. 

@chmst Is this granted forever, or may this change in future? ;-)

**Comment by brianteeman** [comment]
Isnt this the exact use case that @fedik introduced web assets for

**Comment by richard67** [comment]
@brianteeman Not sure if I understand right: You mean web assets are clever enough to handle multiple calls for the same asset so it is included only once, so this PR would not be necessary?

**Comment by brianteeman** [comment]
https://docs.joomla.org/J4.x:Web_Assets

**Comment by Fedik** [comment]
btw, "showon" already exist as an WebAsset item, you can just replace call:
```php
HTMLHelper::_('script', 'system/showon.min.js', array('version' => 'auto', 'relative' => true));
```

to: 
```php
$wa->useScript('showon');
```

and somwhere at top of file:
```php
/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
```

The method `$wa->useScript()` may be called 100 times, it will load the requested script only once, it does not cost performance compared to `HTMLHelper::_('script')` that will search for the script file on each call.

**Comment by chmst** [comment]
Thank you @Fedik. Woud it no be better to load this script in the backend generally? So that no developer must load it if he makes a showon field in her extension?

**Comment by Fedik** [comment]
it will work, but I would prefer to load it only when it need

**Comment by chmst** [comment]
I am confused about the whole showon stuff. 
https://github.com/joomla/joomla-cms/blob/4.0-dev/layouts/joomla/form/renderfield.php#L26

This param "enableShowon" seems to be unknow in the system.

**Comment by chmst** [comment]
Changed loading showon.js as proposed by @Fedik. 
Deleted loading jQuery.

File: components/com_contact/layouts/joomla/form/renderfield.php [hunk 1]
```diff
@@ -9,11 +9,15 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline components/com_contact/layouts/joomla/form/renderfield.php]:
```suggestion
```

File: components/com_users/layouts/joomla/form/renderfield.php [hunk 2]
```diff
@@ -9,11 +9,15 @@
 
 defined('JPATH_BASE') or die;
 
+use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline components/com_users/layouts/joomla/form/renderfield.php]:
```suggestion
```

File: layouts/joomla/form/renderfield.php [hunk 3]
```diff
@@ -9,10 +9,14 @@
 
 defined('JPATH_BASE') or die;
 
+use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
```
Comment (Quy) [inline layouts/joomla/form/renderfield.php]:
```suggestion
```

File: layouts/joomla/searchtools/default/filters.php [hunk 4]
```diff
@@ -14,6 +14,9 @@
```
Comment (Quy) [inline layouts/joomla/searchtools/default/filters.php:14]: Add `use Joomla\CMS\Factory;`
Delete `use Joomla\CMS\HTML\HTMLHelper;`

**Comment by lefabdu51** [comment]
I have tested this item :white_check_mark: successfully on 61bb7aaeffe4f702c2cef07694ceeec112f79122<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27963">issues.joomla.org/tracker/joomla-cms/27963</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 61bb7aaeffe4f702c2cef07694ceeec112f79122<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27963">issues.joomla.org/tracker/joomla-cms/27963</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27963">issues.joomla.org/tracker/joomla-cms/27963</a>.</sub>

**Comment by richard67** [comment]
@chmst Why closing?

**Comment by chmst** [comment]
Sorry, this was a mistake, can you re-open?

**Comment by brianteeman** [comment]
if you have deleted your branch then it cant be reopened

**Pull request closed without merging**